ros2topic/ros2topic/verb/echo.py

```diff
@@ -288,13 +292,15 @@ def _subscriber_callback(self, msg, info):
         if self.future is not None and self.once:
             self.future.set_result(True)
 
+        to_print = ''
+
         if not hasattr(submsg, '__slots__'):
             # raw
             if self.include_message_info:
-                print('---Got new message, message info:---')
-                print(info)
-                print('---Message data:---')
-            print(submsg, end='\n---\n')
+                to_print = '---Got new message, message info:---\n'
+                to_print = f'{to_print}{info}\n'
+                to_print = f'{to_print}---Message data:---\n'
+            print(f'{self.prefix}{to_print}{submsg}', end='\n---\n')
```

Comment:
having escape sequence was what we do in ROS 1 rostopic echo, so this would be probably good.

instead, how about taking advantage of `os.system()` to clear the screen? so that we can just rely on system shell that gives us more flexibility for the future?

```python
import os
import system

if platform.system() == 'Windows':
    os.system('cls')
else:
    os.system('clear')
```

Reply:
I agree that the escape sequence is potentially problematic, particularly on Windows.

That said, I'm really not to enamored of doing `os.system`.  That almost always leads to trouble down the line, as it is difficult to deal with errors and easy to trick it into doing something nefarious.

I think it is worthwhile here to step back and maybe look at some of the python command-line libraries and see what they do in this situation.  If we can find a solution to work on all shells and in Windows (both `cmd` and `powershell`), and does not require `os.system`, that would be the ideal.

Reply:
I think this is a very convenient feature and I would love to see this discussion and PR resolved and merged. Can provide help if it's wanted.
For example, very popular [click](https://github.com/pallets/click/blob/d73083e3960567429428578859d7f7c8fbe0e0a7/src/click/termui.py#L450) python library uses `\033[2J\033[1;1H` escape sequence to clear the screen and reset the cursor position.

Reply:
@Guillaumebeuzeboc @clalancette i just pushed the fix based on the review comment here, can you check?

Reply:
I'm still not a fan of the use of `os.system` in here.  I'd much rather we figure out the correct terminal codes to do this, if we can.

Reply:
The "rich" python library is also using [the escape sequence to clear the screen](https://github.com/Textualize/rich/blob/0f2f51b872d14588de3a65968f5cdde6fb5694a3/rich/control.py#L38).

@clalancette what do you mean by "terminal codes"?
It seems that `os.system` and escape characters are the only solutions that are used around.

Reply:
The [Wikipedia page about ANSI escape code](https://en.wikipedia.org/wiki/ANSI_escape_code#DOS_and_Windows) mentions that escape characters in strings are only supported since PowerShell6 (the default PowerShell for windows10 &11 is 5.1).
I installed power shell 7 and I can confirm that the string ``"`e[2J`e[H"`` is clearing the screen. As you can see, [the escape character is different on Windows](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_special_characters?view=powershell-7.4#escape-e). Also, none of that is going to work natively on Windows since the version of PowerShell is too old.
From here I am really not sure what can be done.

Reply:
![image](https://github.com/user-attachments/assets/2d09af25-3679-4930-b1dd-e15d75b7acf8)
Escape sequence in click library (in `clear()` function) worked perfectly in PowerShell 5.1 for me on Windows 11. Just doing `print("\033[2J\033[1;1H")` works too. So I don't think it's a problem? I think that's an escape character for _strings_ that contain ANSI escape sequences, so you could construct such strings in PowerShell commands. But we don't write PowerShell commands here :)
As Wikipedia article you linked mentions the support for ANSI escape sequences was added and eabled by default since PowerShell 5.1, so I think it's gonna be alright.